### PR TITLE
LPS-60598

### PR DESCRIPTION
**Pull request by neilking**

### CODE REVIEW

File: util-taglib/src/com/liferay/taglib/ui/MessageTag.java [hunk 1]
```diff
@@ -58,11 +58,17 @@ public int doEndTag() throws JspException {
 				if (!_localizeKey) {
 					value = _key;
 				}
-				else if (_escape) {
+				else if (_escape &&
+						 !LanguageUtil.isValidLanguageKey(
+							 resourceBundle.getLocale(), _key)) {
+
 					value = HtmlUtil.escape(
 						LanguageUtil.get(resourceBundle, _key));
```
Comment (samuelkong) [inline util-taglib/src/com/liferay/taglib/ui/MessageTag.java]: This doesn't make sense. Why lookup `_key` in the `resourceBundle` if the key is not a valid language key.

**Comment by neilking** [comment]
it validates if it is a valid key. even it is a valid key, but not exists in resource bundle, it's still invalid.

**Comment by samuelkong** [comment]
> it validates if it is a valid key. even it is a valid key format, but not exists in resource bundle, it's still invalid.

Not sure what you mean. `isValidLanguageKey` doesn't check if it's a valid format. It checks if the key actually exists in the language bundle. So if it's not valid, it's not going to be in the bundle.

**Comment by neilking** [comment]
Right, I check if it is in resource bundle regardless the format.

**Comment by samuelkong** [comment]
You need to also review all uses of `liferay-ui:icon`. After your change, there will be some things that are double escaped.

**Comment by neilking** [comment]
I have reviewed all use of liferay-ui:icon use. Can you tell me what else is double escaped?

**Comment by samuelkong** [comment]
apps/collaboration/bookmarks/bookmarks-web/src/main/resources/META-INF/resources/bookmarks/select_folder.jsp:81

**Comment by neilking** [comment]
Thanks, I will review all of the use again when I have time.

**Comment by neilking** [comment]
I don't think that's a double escape, it's already set localizeMessage="<%= false %>" which means it calls <liferay-ui:message key="<%= message %>" localizeKey="<%= false %>" /> it returns the value directally.

**Comment by samuelkong** [comment]
Based on your changes, the call to 
`message="<%= HtmlUtil.escape(curFolder.getName()) %>"` is wrong. It should just be `message="<%= curFolder.getName() %>"` and `localizeKey="<%= false %>` should be removed.

**Comment by samuelkong** [comment]
Another example:
apps/collaboration/bookmarks/bookmarks-web/src/main/resources/META-INF/resources/bookmarks/select_folder.jsp:77

**Comment by neilking** [comment]
Which means base on my change the liferay-ui:icon tag doesn't need localizeMessage attribute at all in any cases. Do I get you correctly?

**Comment by neilking** [comment]
apps/collaboration/bookmarks/bookmarks-web/src/main/resources/META-INF/resources/bookmarks/select_folder.jsp:81

apps/collaboration/bookmarks/bookmarks-web/src/main/resources/META-INF/resources/bookmarks/select_folder.jsp:77

are the same, right?

**Comment by samuelkong** [comment]
> Which means base on my change the liferay-ui:icon tag doesn't need localizeMessage attribute at all in any cases. Do I get you correctly?

No. It could be needed.

I'm saying `localizeKey="<%= false %>` in select_folder.jsp should be removed.

**Comment by samuelkong** [comment]
My apologizes. Should be apps/collaboration/document-library/document-library-web/src/main/resources/META-INF/resources/document_library/asset/file_entry_abstract.jsp:77

**Comment by neilking** [comment]
localizeMessage is only for localizeKey, if you say localizeKey need to be removed, localizeMessage is not needed anymore. How could it be needed? For legacy to keep an api?

**Comment by neilking** [comment]
And if you do that, in this it would be double escaped anyway.
apps/collaboration/bookmarks/bookmarks-web/src/main/resources/META-INF/resources/bookmarks/select_folder.jsp:84

**Comment by neilking** [comment]
We should provide a place for liferay-ui:icon to turn escape on and off.

**Comment by neilking** [comment]
Are you okey to add a escape attribute to liferay-ui:icon tag and all tags that calls liferay-ui:icon and passed a message?

**Pull request closed without merging**